src/core/opamProcess.ml

```diff
@@ -388,7 +395,9 @@ let create ?info_file ?env_file ?(allow_stdin=not Sys.win32) ?stdout_file ?stder
     | Some f ->
       let chan = open_out f in
       let info =
-        make_info ~cmd ~args ~cwd ~env_file ~stdout_file ~stderr_file ~metadata () in
+        make_info ~cmd ~args ~cwd ~env_file ~stdout_file ~stderr_file
+          ~metadata:(Lazy.force metadata) ()
```

Comment:
this is forced too early. I think there is a way to run it only when the process finished